RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate.cc

```diff
@@ -134,36 +134,36 @@ bool SiPixelTemplate::pushfile(int filenum, std::vector< SiPixelTemplateStore >
    // Local variables
    int i, j, k, l;
    float qavg_avg;
-   const char *tempfile;
-   //	char title[80]; remove this
    char c;
    const int code_version={17};
    
-   
-   
    //  Create a filename for this run
-   
-   std::ostringstream tout;
-   
-   //  Create different path in CMSSW than standalone
+   std::string tempfile = std::to_string(filenum);
    
 #ifndef SI_PIXEL_TEMPLATE_STANDALONE
-   tout << dir << "template_summary_zp"
-	<< std::setw(4) << std::setfill('0') << std::right << filenum << ".out" << std::ends;
-   std::string tempf = tout.str();
-   edm::FileInPath file( tempf.c_str() );
-   tempfile = (file.fullPath()).c_str();
+   // If integer filenum has less than 4 digits, prepend 0's until we have four numerical characters, e.g. "0292"
+   int nzeros = 4-tempfile.length();
+   if (nzeros > 0)
+     tempfile = std::string(nzeros, '0') + tempfile;
```

Comment:
why not create tempfile name (which should  be `template_summary_zpNNNN.out`) once before the `#ifndef` condition and just add directory to it if SI_PIXEL_TEMPLATE_STANDALONE is not defined?

Reply:
@smuzaffar 
Sure, we can do that, but then we need to agree on the common logic to add strings.  If using ostringstream works on gcc 7X, then let's go with that.  Otherwise we'll need to negotiate.

The macro SI_PIXEL_TEMPLATE_STANDALONE is of course not defined in CMSSW.  (Because CMSSW is not standalone -- standalone is when one is without CMSSW.)   The point is that once you define it, the code can run *outside* CMSSW and we can quickly develop and test new features based on PixelTrees (which have calibrated ADC counts and can be used for local reco)!  So we most definitely do need the #else part.